test/Microsoft.AspNetCore.Razor.Language.Test/Legacy/CSharpDirectivesTest.cs

```diff
@@ -11,6 +11,104 @@ namespace Microsoft.AspNetCore.Razor.Language.Legacy
     public class CSharpDirectivesTest : CsHtmlCodeParserTestBase
     {
         [Fact]
+        public void ExtensibleDirectiveDoesNotErorrIfNotAtStartOfLineBecauseOfWhitespace()
+        {
+            // Arrange
+            var descriptor = DirectiveDescriptor.CreateDirective(
+                "custom",
+                DirectiveKind.SingleLine,
+                b => b.AddTypeToken());
+
+            // Act & Assert
+            ParseCodeBlockTest(Environment.NewLine + "  @custom System.Text.Encoding.ASCIIEncoding",
+                new[] { descriptor },
+                new DirectiveBlock(
+                    new DirectiveChunkGenerator(descriptor),
+                    Factory.Code(Environment.NewLine + "  ").AsStatement(),
+                    Factory.CodeTransition(),
+                    Factory.MetaCode("custom").Accepts(AcceptedCharactersInternal.None),
+                    Factory.Span(SpanKindInternal.Code, " ", markup: false).Accepts(AcceptedCharactersInternal.WhiteSpace),
+                    Factory.Span(SpanKindInternal.Code, "System.Text.Encoding.ASCIIEncoding", markup: false)
+                        .With(new DirectiveTokenChunkGenerator(descriptor.Tokens[0]))
+                        .Accepts(AcceptedCharactersInternal.NonWhiteSpace)));
+        }
+
+        [Fact]
+        public void BuiltInDirectiveDoesNotErorrIfNotAtStartOfLineBecauseOfWhitespace()
+        {
+            // Act & Assert
+            ParseCodeBlockTest(Environment.NewLine + "  @addTagHelper \"*, Foo\"",
+                Enumerable.Empty<DirectiveDescriptor>(),
+                new DirectiveBlock(
+                    Factory.Code(Environment.NewLine + "  ").AsStatement(),
+                    Factory.CodeTransition(),
+                    Factory.MetaCode(SyntaxConstants.CSharp.AddTagHelperKeyword + " ")
+                            .Accepts(AcceptedCharactersInternal.None),
+                    Factory.Code("\"*, Foo\"")
+                        .AsAddTagHelper("\"*, Foo\"")));
+        }
+
+        [Fact]
+        public void BuiltInDirectiveErorrsIfNotAtStartOfLine()
```

Comment:
Is this a breaking change? We don't want to just make arbitrary breaking changes without agreement. This will affect users of 1.X as well.